lib/components/steps/mapping/MappingSearch.tsx

```diff
@@ -2,31 +2,32 @@ import React, {useState} from "react";
 import {Box, Button, InputAdornment, TextField} from "@mui/material";
 import {FilterIcon, SearchIcon} from "../../../icons";
 import Filters from "../../common/Filters.tsx";
+import { useDebounce } from "../../../hooks.ts";
 
 
 interface MappingSearchProps {
-    onChange: () => void;
+    onChange: (searchTerm: string) => void;
 }
 
 export default function MappingSearch({onChange}: MappingSearchProps) {
 
     const [searchString, setSearchString] = useState('');
 
     const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
-
-
-    const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-        setSearchString(event.target.value);
-        onChange()
-    };
+    const debouncedSearchValue = useDebounce(searchString);
 
     const handleFiltersClose = () => {
         setAnchorEl(null);
-        onChange()
+        onChange(debouncedSearchValue)
```

Comment:
As discussed in slack, is is most likely incomplete but it's out of the scope of the current PR to implement the filtering. For the filtering integration we will need to know what filters are active and have the onChange filter the rows accordingly.